Lib/test/test_defaultdict.py

```diff
@@ -183,5 +183,30 @@ def test_pickling(self):
             o = pickle.loads(s)
             self.assertEqual(d, o)
 
+    def test_union(self):
+        i = defaultdict(int, {1: 1, 2: 2})
+        s = defaultdict(str, {0: "zero", 1: "one"})
+
+        self.assertIs((i | s).default_factory, int)
+        self.assertDictEqual(i | s, {1: "one", 2: 2, 0: "zero"})
```

Comment:
Is defaultdict not order-preserving, like dict? If it is, could the test check it?

Reply:
You found a flaw in my update logic! I pushed a fix and added tests for key order.